reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -216,8 +216,11 @@ object StreamSubscriber {
       def onComplete(): Unit = nextState(OnComplete)
       def onFinalize: F[Unit] = F.delay(nextState(OnFinalize))
       def dequeue1: F[Either[Throwable, Option[Chunk[A]]]] =
-        F.async_[Either[Throwable, Option[Chunk[A]]]] { cb =>
-          nextState(OnDequeue(out => cb(Right(out))))
+        F.async[Either[Throwable, Option[Chunk[A]]]] { cb =>
+          F.delay {
+            nextState(OnDequeue(out => cb(Right(out))))
+            Some(F.unit)
```

Comment:
The cleanup happens with the `onFinalize` call in the `Streams.bracket` here:
```scala
def stream(subscribe: F[Unit])(implicit ev: ApplicativeError[F, Throwable]): Stream[F, A] =
      Stream.bracket(subscribe)(_ => onFinalize) >> Stream
        .eval(dequeue1)
        .repeat
        .rethrow
        .unNoneTerminate
        .unchunks
```
which is the only place where `dequeue1` is used internally.

Although technically I guess someone could call `StreamSubscriber#sub.dequeue1` directly, since it is exposed here (unlike analogous `flow` interop).